src/twisted/test/test_task.py

```diff
@@ -506,10 +506,10 @@ def foo(cnt):
         loop.clock = clock
         deferred = loop.start(0, now=False)
 
-        clock.advance(0)
+        clock.pump([0] * 5)
```

Comment:
I guess we can also have the test like this

And my understanding is that this is is how the tests will be refactored after the Clock is fixed.

```suggestion
        # Iterating the reactor will only cause the loop to loop once.
        # even if the loop is configured to loop without a delay.
        clock.advance(0)
        
        self.assertEqual([1], accumulator)
```

Reply:
Because of the assertion on line 512 (accumulator == [1] * 5) the clock/reactor will still need to iterate 5 times.  It is the old, broken behavior that makes `clock.advance(0)` run the LoopingCall 5 times.